### PR TITLE
👶 steps for comms

### PR DESCRIPTION
**Pull request by romainfrancois**
related to #43 

 - Adds xeusr_comm_register_target and xeusr_comm_unregister_target for now. 
 - Needs comm_open(), comm_close() ...

### CODE REVIEW

File: src/routines.cpp [hunk 1]
```diff
@@ -74,6 +79,56 @@ SEXP xeusr_log(SEXP level_, SEXP msg_) {
...
+
+    auto content = nl::json {
+        {"comm_id", comm_id}, 
+        {"target_name", target_name}, 
+        {"data", data}
+    };
+
+    auto msg = xeus::xmessage(
+        /* zmq_id = */        {},                  // TODO: not sure where to get `zmq_id` from
+        /* header = */        nl::json::object(),  // TODO: what should this be ?
```
Comment (romainfrancois) [inline src/routines.cpp:120]: ping @SylvainCorlay I think I'm on the right track here, but 🤷

File: src/routines.cpp [hunk 2]
```diff
@@ -74,6 +79,56 @@ SEXP xeusr_log(SEXP level_, SEXP msg_) {
...
+}
+
+SEXP comm_manager__unregister_target(SEXP name_) {
+    std::string name = CHAR(STRING_ELT(name_, 0));
+
+    xeus_r::get_interpreter()->comm_manager().unregister_comm_target(name);
+    return R_NilValue;
+}
+
+SEXP comm_manager__comm_open(SEXP s_comm_id, SEXP s_target_name, SEXP js_data) {
```
Comment (romainfrancois) [inline src/routines.cpp:105]: Perhaps the comm_id should not come from the R side and be generated automatically with `new_xguid` ?

**Comment by SylvainCorlay** [comment]
It seems that this PR broke the windows build.

**Comment by SylvainCorlay** [comment]
> C:\Users\runneradmin\micromamba-root\envs\xeus-r\Lib\R\include\R_ext/Complex.h(81): error C3646: 'private_data_c': unknown override specifier

**Comment by romainfrancois** [comment]
That's odd. I haven't touched `Complex.h` things. I'll have a look.

**Comment by anutosh491** [comment]
I attempted to fix the windows build and have written my logic here (https://github.com/jupyter-xeus/xeus-r/pull/100#issuecomment-2363978030). The build & tests pass as expected.

cc @SylvainCorlay @romainfrancois

**Pull request merged**